components/common/src/main/java/com/hotels/styx/server/HttpInterceptorContext.java

```diff
@@ -17,18 +17,54 @@
 
 import com.hotels.styx.api.HttpInterceptor;
 
+import java.net.InetSocketAddress;
 import java.util.Map;
+import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * ConcurrentHashMap backed implementation of HttpInterceptor.Context.
  */
 public final class HttpInterceptorContext implements HttpInterceptor.Context {
     private final Map<String, Object> context = new ConcurrentHashMap<>();
-    private boolean secure;
 
-    private HttpInterceptorContext(boolean secure) {
+    private final boolean secure;
+    private final InetSocketAddress clientAddress;
+
+    /**
+     * Construct a new instance.
+     *
+     * @param secure true if the request was received via SSL
+     * @param clientAddress address that request came from, or null if not-applicable
+     */
+    public HttpInterceptorContext(boolean secure, InetSocketAddress clientAddress) {
```

Comment:
Please back off this change to the constructor and the static factory method. This PR is now changing 53 files. Majority of them simply because of this unrelated constructor change. Let's keep this PR strictly to removing the client address. This way it is easier to justify and manage the changes, and to review the PRs.

Whether or not it was good idea back in the days to hide the constructor in favour of static factory method is another matter. If you feel strongly about it then please propose that in a separate pull request.

This change will also invalidate any plugins that use `HttpInterceptorContext.createa in unit tests.

Reply:
I have added back the method `HttpInterceptorContext.create()` and reverted tests that use the new constructor.